example/test_problem/Hydro/CCSN/python_script/convert_progenitor_form.py

```diff
@@ -0,0 +1,164 @@
+#! /usr/bin/env python
+#
+# -*- coding: utf-8 -*-
+#
+#  Purpose:
+#    Converting the progenitor file format (WH07, Heager2005, Sukhbold2015, and MESA)
+#    to the format that GAMER can use for CCSN test problems
+#
+
+import argparse
+import sys
+import numpy as np
+
+
+def convert_WHS(file):
+
+    # read file info
+    f_info = file.readline()
+
+    # read the header
+    headers = []
+    line_header = file.readline()
+    headers = [word.strip() for word in line_header.strip().split("  ") if word]
+
+    for i in range(len(headers)):
+        keys = ["cell outer radius", "cell density", "cell temperature", "cell outer velocity",
+                "cell Y_e", "cell pressure", "cell angular velocity"]
+
+    # find and read targeting variables (radius, density, temperature, radial velocity,
+    #                                    electron fraction, pressure, and angular velocity)
+    nvars = 7
+    TargetCols = np.empty(nvars, dtype=np.int)
+    for i in range(len(headers)):
+        keys = ["cell outer radius", "cell density", "cell temperature", "cell outer velocity",
+                "cell Y_e", "cell pressure", "cell angular velocity"]
+    TargetCols = [headers.index(key) for key in keys]
+
+    # GAMER-formatted array
+    GAMER_DATA_FORM = np.empty((0, nvars), dtype=np.float)
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    outer_radius = 0.0
+    outer_vel    = 0.0
+    for line in f:
+        if ('wind:' in line):
+            break
+
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line.split()[icol])
+            # convert cell outer variables to cell centered values (radius, velocity)
+            if idx == 0:
+                variables[idx], outer_radius = 0.5 * (outer_radius + variables[idx]), variables[idx]
+            if idx == 3:
+                variables[idx], outer_vel = 0.5 * (outer_vel + variables[idx]), variables[idx]
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+def convert_MESA(file):
+
+    # physical constants
+    R_sol = 6.977e10 # solar radius
```

Comment:
Is this value adopted from MESA?

Reply:
The exact value is `6.957e10`. I updated the value.